### ISSUE TITLE
Intermittent/recurring issue - Users seeing repeated build failures - Jenkins slave never connects - BZ# 1552373

### ISSUE THREAD

**Issue by ldimaggi** [openshiftio/openshift.io#2427]
Seeing this error on approx. 40% of build pipelines for user account provisioned on cluster starter-us-east-2a

Steps to recreate:

* Reset user env
* Create new space and quickstart
* Observe Jenkins pod log

Seeing this sequence:

Jenkins slave fails to start after 100 tries:
```
Mar 03, 2018 10:30:43 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesLauncher launch
INFO: Waiting for slave to connect (99/100): jenkins-slave-x66ct-3tjc3
Mar 03, 2018 10:30:44 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesLauncher logLastLines
SEVERE: Error in provisioning; slave=KubernetesSlave name: jenkins-slave-x66ct-3tjc3, template=org.csanchez.jenkins.plugins.kubernetes.PodTemplate@162a824. Container jnlp. Logs: Using 64 bit Java since OPENSHIFT_JENKINS_JVM_ARCH is not set
Downloading http://jenkins//jnlpJars/remoting.jar  ...
curl: (7) Failed connect to jenkins:80; No route to host

Mar 03, 2018 10:30:44 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesLauncher launch
WARNING: Error in provisioning; slave=KubernetesSlave name: jenkins-slave-x66ct-3tjc3, template=org.csanchez.jenkins.plugins.kubernetes.PodTemplate@162a824
java.lang.IllegalStateException: Slave is not connected after 100 attempts, status: Running
```
Then - it tries again:
```
Mar 03, 2018 10:30:58 PM okhttp3.internal.platform.Platform log
INFO: ALPN callback dropped: HTTP/2 is disabled. Is alpn-boot on the boot class path?
Mar 03, 2018 10:30:58 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesLauncher launch
INFO: Created Pod: jenkins-slave-x66ct-h9gdm in namespace ldimaggi-osiotest3-jenkins
Mar 03, 2018 10:30:58 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesLauncher launch
INFO: Waiting for Pod to be scheduled (0/100): jenkins-slave-x66ct-h9gdm
Mar 03, 2018 10:31:04 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesLauncher launch
INFO: Waiting for slave to connect (0/100): jenkins-slave-x66ct-h9gdm
Mar 03, 2018 10:31:05 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesLauncher launch
INFO: Waiting for slave to connect (1/100): jenkins-slave-x66ct-h9gdm
```
And - quotas are exceeded:
```
https://172.30.0.1/api/v1/namespaces/ldimaggi-osiotest3-jenkins/pods . Message:
Forbidden!Configured service account doesn't have access. Service account may have been revoked
pods "jenkins-slave-x66ct-cgvcv" is forbidden: exceeded quota: compute-resources-timebound,
requested: limits.cpu=1750m,limits.memory=896Mi, used: limits.cpu=3500m,limits.memory=1792Mi,
limited: limits.cpu=4,limits.memory=2Gi.
```
The Jenkins pod at this point is in a loop - the "Service account may have been revoked" is repeated every 10 seconds - and the user's build pipeline is never able to perform a build. The user must reset the environment and start over. 

See ldimaggi for the full Jenkins log.

![screenshot from 2018-03-03 18-00-07](https://user-images.githubusercontent.com/642621/36940226-eb15919e-1f0c-11e8-8d1f-1518eedfdd5d.png)

![screenshot from 2018-03-03 18-17-38](https://user-images.githubusercontent.com/642621/36940340-303f951a-1f0f-11e8-8a9d-786b92fd948c.png)

**Comment by ldimaggi** [comment]
Issue found/encountered by EE tests

**Comment by kbsingh** [comment]
Something for the build team to investigate, let us know if we can help in anyway from SD

**Comment by jaseemabid** [comment]
All of my builds failed with the same issue. This needs investigation.

**Comment by ldimaggi** [comment]
Are you seeing ALL (100%) of your builds fail with this error? 

If we are seeing all users on cluster starter-us-east-2a unable to create/execute a build pipeline, then I think that we should designate this issue as SEV1.

**Comment by jaseemabid** [comment]
I tried a couple of quickstarts, ran a few builds per app and all of them failed with `Jenkins doesn’t have label mavenvertx2_3`. We've known/seen this for months. It just got unbearable on the new cluster. 

AFAIK, its getting stuck in this [loop](https://github.com/jenkinsci/kubernetes-plugin/blob/master/src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java#L118) in the kubernetes-plugin and never recovering in a clean slate. 

Should probably be a SEV1. Now fixing this involves digging deep into the kubernetes-plugin and none of us know that code well enough. This will take time to understand and fix, irrespective of the labels. 

Ref https://github.com/fabric8io/fabric8-build-team/issues/17

**Comment by ldimaggi** [comment]
@jaseemabid   What are you seeing in the Jenkins pod log at this time? Is it caught in a 10-second loop trying to connect but failing due to quota being exceeded?

**Comment by jfchevrette** [comment]
I've raised an issue upstream: https://bugzilla.redhat.com/show_bug.cgi?id=1552373

This seems to be a dupe / same root cause as #2426

**Comment by ldimaggi** [comment]
The upstream bug is still open - but I am not seeing this issue on cluster starter-us-east-2a this weekend.

**Comment by ldimaggi** [comment]
When this issue was originally logged, the problem was only seen on the starter-us-east-2a cluster.

On March 14 and 15, George Acton and I started seeing it on the starter-us-east-2 cluster. The problem is only being seen randomly/intermittently. When it occurs, the pattern is consistent with that described in this issue.

**Comment by GeorgeActon** [comment]
Saw this again today
After 100 tries
then seemed to be starting on the seocnd series of tries on try 7
but failed

**Comment by ldimaggi** [comment]
It looks like after 2 hours of trying, the build will be marked as failed in the OSIO UI.

**Comment by chmouel** [comment]
I have seen the same issue tonight and so is some people, from my investigation the jnlp interface doesn't get open : 

![image](https://user-images.githubusercontent.com/98980/37479820-dc1a1c26-287d-11e8-9c28-f8e3180c88d2.png)

cause it waits to be able to download something from http://jenkins

the problem the jenkins internal service doesn't seems to  be available between pods : 

![image](https://user-images.githubusercontent.com/98980/37479862-fccb3676-287d-11e8-8dab-a6385d81d67c.png)

while working from jenikins pod : 

![image](https://user-images.githubusercontent.com/98980/37479903-1f502bf2-287e-11e8-86ff-e007086d066c.png)

I am going to set as SEV1 since this is getting really worrysome,

@jfchevrette @kbsingh @pbergene

**Comment by sthaha** [comment]
This is my observation about issue which I performed on `cluster 2a`. 

Initially when jenkins master pod starts it takes a while (2 - 3 mins) to be fully functional during which the jenkins server return 504 but the host itself was reachable from outside. One jenkins settled down. All my 3 builds succeeded without any issues.

Note sure if this is related; my account on cluster-2a got activated only today and jenkins image it is running is - `fabric8/jenkins-openshift:v94386a0`

**Comment by chmouel** [comment]
@staha (or anyone else) if you do some investigations please investigates by giving the log output of the pods like done on my screenshots previously. The cause seems to be of issues with intercommunication between the pods and we should look her first.

I had a theory at first that maybe since we have deployed a new idler yesterday, maybe it increased the network throughput and then brings down (or slow it by a lot) the openshift sdn. Chatting with @hferentschik and pulling some datas from the monitoring side, idler didn't seem to do any extra traffic before and after the deployed version.

We have been requested to provide a clear path to reproduce this but that's not something no one of us has been able to do. It seems to happen (and that's a theory since we don't have any control or view on what's happening on openshift globally) that when we have a lot of network traffic happening on the cluster, the jenkins-slave is not able to contact the master anymore and then things fails.

I don't think there is much at this point we can do from our line of sight and we need to have some help from the ops teams.

**Comment by chmouel** [comment]
I am not seeing any failed build at the moment on my account : 

![image](https://user-images.githubusercontent.com/98980/37523302-cfd84632-2926-11e8-8f26-107173ac5a0e.png)

I am going to get this issue down to SEV-2 since thi sis not impacting production anymore but I am hoping osops can follow with at least some network graph of the platform so we can see what's going on,

**Comment by chmouel** [comment]
If you ever get this issue please provide the logs of the `jenkins-master`, the `jenkins-slave` with both `jnlp` and `maven` containers and connect to the content-repository pod and try to do a curl -v http://jenkins/

**Comment by ldimaggi** [comment]
Saw this issue on Monday, March 19 - seeing this on the content repo pod:

```
oc rsh content-repository-1-5m28r

sh-4.2$ curl -v http://jenkins 

* About to connect() to jenkins port 80 (#0)
*   Trying 172.30.123.123...
* Connected to jenkins (172.30.123.123) port 80 (#0)
> GET / HTTP/1.1
> User-Agent: curl/7.29.0
> Host: jenkins
> Accept: */*
> 
< HTTP/1.1 403 Forbidden
< Date: Mon, 19 Mar 2018 17:58:34 GMT
< X-Content-Type-Options: nosniff
< Set-Cookie: JSESSIONID.4a8aa965=node0qi1g80go912aunr1fdzspu8p107.node0;Path=/;HttpOnly
< Expires: Thu, 01 Jan 1970 00:00:00 GMT
< Content-Type: text/html;charset=utf-8
< X-Hudson: 1.395
< X-Jenkins: 2.93
< X-Jenkins-Session: f21f97fd
< X-Hudson-CLI-Port: 50000
< X-Jenkins-CLI-Port: 50000
< X-Jenkins-CLI2-Port: 50000
< X-You-Are-Authenticated-As: anonymous
< X-You-Are-In-Group-Disabled: JENKINS-39402: use -Dhudson.security.AccessDeniedException2.REPORT_GROUP_HEADERS=true or use /whoAmI to diagnose
< X-Required-Permission: hudson.model.Hudson.Read
< X-Permission-Implied-By: hudson.security.Permission.GenericRead
< X-Permission-Implied-By: hudson.model.Hudson.Administer
< Content-Length: 837
< Server: Jetty(9.4.z-SNAPSHOT)
< 
<html><head><meta http-equiv='refresh' content='1;url=/securityRealm/commenceLogin?from=%2F'/><script>window.location.replace('/securityRealm/commenceLogin?from=%2F');</script></head><body style='background-color:white; color:white;'>


Authentication required
<!--
You are authenticated as: anonymous
Groups that you are in:
  
Permission you need to have (but didn't): hudson.model.Hudson.Read
 ... which is implied by: hudson.security.Permission.GenericRead
 ... which is implied by: hudson.model.Hudson.Administer
-->

* Connection #0 to host jenkins left intact
```

</body></html>

**Comment by chmouel** [comment]
@ldimaggi  I would believe there is something else at play here since you do seem to have inter pod networking here,

**Comment by ldimaggi** [comment]
I was seeing this error repeatedly late yesterday, but only on the first cluster.

**Comment by jfchevrette** [comment]
The clusters have been upgraded to 3.9 and the openshift team believe this issue may have been fixed by some of the SDN/networking patches that were included in 3.9.

@chmouel @ldimaggi if you still see this issue can you please reach out to me directly on chat.openshift.io so I can do live troubleshooting?

**Comment by ldimaggi** [comment]
Closing as not reproducible.
